Automattic/jetpack
# Changes the error Blaze shows during sync to be a warning

We want to improve the user's first experience when connecting to their site and then navigating to the Blaze Dashboard. 

Because Blaze depends on the posts being synced to work properly, we now have a blocking error that will warn the user that they need to wait until the sync finishes to start promoting their content. 

This is partially true because new content will get synced when it is created and can be promoted immediately.

## Proposed changes:
* Changes the current error in the `/sites/:id/blaze/posts`endpoint to be a warning instead of a blocking error.

We will merge a Calypso compatibility PR that will handle that new warning before we merge this PR.
Calypso PR: https://github.com/Automattic/wp-calypso/pull/94862

### Other information:


## Jetpack product discussion


N/A

## Does this pull request change what data or activity we track or use?


No

## Testing instructions:






* Create a new ephemeral site (or any new site), and connect it with this PR version of Jetpack
* Navigate to the Advertising Dashboard (Tools->Advertising) and verify that you see the warning, but also are able to see the search bar
![Screenshot 2024-09-27 at 10 53 09 AM](https://github.com/user-attachments/assets/7f184642-8934-44d2-ac16-7a5cb49c06b4)
* If you create a new post, you should see the warning message and the new post on the list.
![Screenshot 2024-09-27 at 10 53 48 AM](https://github.com/user-attachments/assets/4a7fcc39-02cf-43bd-80ed-1dffead12223)


**Note:** The `sync_in_progress` warning will only be active if the site is synced. It usually takes a little time (~5 to 10 minutes) for new sites.
